src/crimson/net/Dispatcher.h

```diff
@@ -16,40 +16,77 @@
 
 #include <seastar/core/future.hh>
 #include <seastar/core/sharded.hh>
+#include <boost/intrusive/slist.hpp>
 
+#include "crimson/common/gated.h"
 #include "Fwd.h"
 
 class AuthAuthorizer;
 
 namespace crimson::net {
 
-class Dispatcher {
+class Dispatcher : public crimson::common::Gated,
+		   public boost::intrusive::slist_base_hook<
+			    boost::intrusive::link_mode<
+			      boost::intrusive::safe_link>> {
  public:
+  Dispatcher() = default;
+  Dispatcher(Dispatcher&&) = default;
   virtual ~Dispatcher() {}
 
-  virtual seastar::future<> ms_dispatch(Connection* conn, MessageRef m) {
-    return seastar::make_ready_future<>();
+  seastar::future<> ms_dispatch(Connection* conn, MessageRef m) {
+    return gated_dispatch_with_future(__func__, [this, conn, m] {
+      return _ms_dispatch(conn, m);
```

Comment:
_ms_dispatch (and friends) are an interface between Dispatcher and its inheritor.  Can we give them a more descriptive name so readers of the inheritors have a hint as to where to find the user?  Perhaps ms_chained_dispatch.